### ISSUE TITLE
Add a manpage backend

### ISSUE THREAD

**Issue by pjanx** [bytesparadise/libasciidoc#793]
https://asciidoctor.org/docs/user-manual/#man-pages

A subset is highly useful to document almost everything: bold, italics, links, section headers, definition lists, literal blocks.

Apparently relates to #581.

**Comment by pjanx** [comment]
I almost have an MVP. Currently blocking on #800.

**Comment by xcoulon** [comment]
> I almost have an MVP. Currently blocking on #800.

oh, a PR for manpage support would be great! 🤩

**Comment by pjanx** [comment]
Yeah, I'm in the situation where 80% of time will be spent on 20% of details. I'm trying to make it as small as possible.

**Comment by pjanx** [comment]
If you want to help, this needs to be abstracted away, manpages have `.[1-9][a-z]` extensions defined by section 0 heading:
```
		outname := strings.TrimSuffix(path, filepath.Ext(path)) + ".html"
```
The filename isn't known until parsing (or generation). What do?

**Comment by xcoulon** [comment]
> If you want to help, this needs to be abstracted away, manpages have `.[1-9][a-z]` extensions defined by section 0 heading:
> 
> ```
> 		outname := strings.TrimSuffix(path, filepath.Ext(path)) + ".html"
> ```
> 
> The filename isn't known until parsing (or generation). What do?

I'm not sure I'm following you. Maybe you could create a PR in draft mode and point me to where you have blockers?

**Comment by pjanx** [comment]
foo.adoc:
```
= foo(1)
== Name
foo - does things
```
```
asciidoctor -b manpage foo.adoc
```
This produces file _foo.1_. The `Convert` function is an unusable interface (also for EPUB or whatever).

**Comment by pjanx** [comment]
Well, it could probably still be used if the extension were placed in the Metadata struct but you'd have to buffer the output or create a temporary one up first (which isn't a bad idea anyway, to use atomic rewrites).

**Comment by xcoulon** [comment]
> foo.adoc:
> 
> ```
> = foo(1)
> == Name
> foo - does things
> ```
> 
> ```
> asciidoctor -b manpage foo.adoc
> ```
> 
> This produces file _foo.1_. The `Convert` function is an unusable interface (also for EPUB or whatever).

ok, I see what you mean now. Or you could use the `-o` flag to specify the output, maybe?

**Comment by xcoulon** [comment]
> Well, it could probably still be used if the extension were placed in the Metadata struct but you'd have to buffer the output or create a temporary file up first (which isn't a bad idea anyway, to use atomic rewrites).

I'm fine with adding this kind of feature for manpage docs, it shouldn't be too hard, as you suggested, to buffer or write in a temp file and rename it afterwards

**Comment by pjanx** [comment]
I'll return to this sooner or later but right now I'm postponing it.

**Comment by pjanx** [comment]
If anyone happens to need this feature, I've impulsively created [a trivial AsciiDoc to man page converter in AWK](https://git.janouch.name/p/liberty/src/branch/master/tools/asciiman.awk) for a basic subset of the language (I'm using it as a fallback for [1](https://git.janouch.name/p/json-rpc-shell/src/branch/master/json-rpc-shell.adoc), [2](https://git.janouch.name/p/xK/src/branch/master/xB.adoc), [3](https://git.janouch.name/p/xK/src/branch/master/xC.adoc), [4](https://git.janouch.name/p/xK/src/branch/master/xD.adoc), [5](https://git.janouch.name/p/nncmpp/src/branch/master/nncmpp.adoc)). Perhaps it works better than it should.

Thus, even though it should remain fairly easy to teach libasciidoc how to do it properly, I'm no longer particularly invested.
